modules/core/src/local_pool.cpp

```diff
@@ -0,0 +1,55 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "opencv2/core/utils/local_pool.private.hpp"
+#include "opencv2/core/utils/configuration.private.hpp"
+
+#ifdef OPENCV_ENABLE_MEMORY_SANITIZER
+#define LOCAL_POOL_DEFAULT_MODE true
+#else
+#define LOCAL_POOL_DEFAULT_MODE false
+#endif
+
+static bool CV_LOCAL_POOL_OVERRIDE_SAFE_MODE =
+    cv::utils::getConfigurationParameterBool("OPENCV_LOCAL_POOL_ALWAYS_SAFE", LOCAL_POOL_DEFAULT_MODE);
+
+namespace cv { namespace utils {
+
+LocalPool::LocalPool(bool safe_)
+    : safe(safe_ || CV_LOCAL_POOL_OVERRIDE_SAFE_MODE),
```

Comment:
We need a way to turn off this mode using environment (may be useful if problem can't be reproduced with switched ON mode).

Reply:
We assume that nobody will create an object with safe mode enabled `LocalPool::LocalPool(true)`, it is available mainly for testing purpose.